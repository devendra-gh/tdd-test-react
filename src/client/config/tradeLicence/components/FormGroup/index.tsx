import React, { useCallback, useState } from 'react';
import { IVariables } from '@tamm/app-composer';
import Form from '@tamm/ui-lib-v2-form';
import AccordionItem from '@tamm/ui-lib-v2-accordion/AccordionItem/AccordionItem';
import useMedia from 'use-media';

function FormGroup(props: IVariables) {
  const [expended, setExpended] = useState(false);
  const isWide = useMedia({ minWidth: '768px' });

  const onToggle = useCallback(() => {
    setExpended(!expended);
  }, [expended]);

  if (!props.visible) {
    return null;
  }

  const x: any = {
    name: props.name,
  };

  const content = (
    <div>
      <a {...x}> </a>
      {props.children}
    </div>
  );

  if (isWide) {
    return <Form.Fieldset title={props.title}>{content}</Form.Fieldset>;
  }

  return (
    <AccordionItem
      id={props.name}
      title={props.title}
      boldTitle
      expanded={expended}
      onClick={onToggle}
    >
      {content}
    </AccordionItem>
  );
}

FormGroup.defaultProps = {
  visible: true,
};

export default FormGroup;
