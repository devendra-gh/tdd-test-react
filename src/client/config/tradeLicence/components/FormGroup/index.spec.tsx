import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { Viewport, ViewportProvider } from '@tamm/ui-lib-v2-viewport';
import '@testing-library/jest-dom/extend-expect';
import useMedia from 'use-media';
import FormGroup from './index';

jest.mock('use-media');

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Components/FormGroup', () => {
  let mediaMock: any;
  beforeEach(() => {
    mediaMock = useMedia;
  });

  afterEach(cleanup);

  it('should properly render FormGroup when not visible', () => {
    const props = {
      visible: false,
    };

    render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <FormGroup {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
  });

  it('should properly render FormGroup with wide', () => {
    const props = {
      visible: true,
    };
    mediaMock.mockReturnValue(true);

    render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <FormGroup {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
  });

  it('should properly render FormGroup without wide', () => {
    const props = {
      visible: true,
    };
    mediaMock.mockReturnValue(false);

    const { container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <FormGroup {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
    // fireEvent.click(getByText('Add Outlined Icon'));
  });
});
