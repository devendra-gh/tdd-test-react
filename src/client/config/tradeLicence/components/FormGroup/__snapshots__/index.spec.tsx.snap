// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FormGroup should properly render FormGroup without wide 1`] = `
<div>
  <div
    class="ui-lib-viewport ui-lib-viewport_lg"
  >
    <div
      class="ui-lib-accordion-item"
      data-no-side-paddings="false"
      data-ui-type="default"
    >
      <button
        class="ui-lib-accordion-item__head"
        type="button"
      >
        <div
          class="ui-lib-accordion-item__head-title"
          data-bold="true"
          data-wide="true"
        />
        <div
          class="ui-lib-accordion-item__head-indicator-wrapper"
        >
          <svg
            class="ui-lib-accordion-item__head-indicator ui-lib-icon"
            data-key=""
            height="24"
            tabindex="0"
            version="1.1"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
            />
          </svg>
        </div>
      </button>
      <div
        class="ui-lib-accordion-item__body"
        data-expanded="false"
        style="max-height: 0; overflow: hidden;"
      >
        <div
          class="ui-lib-accordion-item__body-content"
        >
          <div>
            <a>
               
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
